packages/dd-trace/src/appsec/stack_trace.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const { calculateDDBasePath } = require('../util')
+
+const ddBasePath = calculateDDBasePath(__dirname)
+
+const LIBRARY_FRAMES_BUFFER = 20
+
+function getCallSiteList (maxDepth = 100) {
+  const previousPrepareStackTrace = Error.prepareStackTrace
+  const previousStackTraceLimit = Error.stackTraceLimit
+  let callsiteList
+  Error.stackTraceLimit = maxDepth
+
+  try {
+    Error.prepareStackTrace = function (_, callsites) {
+      callsiteList = callsites
+    }
+    const e = new Error()
+    e.stack
+  } finally {
+    Error.prepareStackTrace = previousPrepareStackTrace
+    Error.stackTraceLimit = previousStackTraceLimit
+  }
+
+  return callsiteList
+}
+
+function filterOutFramesFromLibrary (callSiteList) {
+  return callSiteList.filter(callSite => !callSite.getFileName().includes(ddBasePath))
+}
```

Comment:
would StartsWith also work ?
```suggestion
  return callSiteList.filter(callSite => !callSite.getFileName().startsWith(ddBasePath))
}
```

Reply:
Yes, I think it will work as `includes`. What's the reason to change it? Is it more performant?

Reply:
Well yeah, if it works just as well, then instead of checking the substring in every possible position of the string, it's only checking the first position. You change a O(n) into a O(1) basically